# Merging DataFrames for EDA#

In [1]:
import pandas as pd

In [37]:
!pwd

/Users/italo/Desktop/Course Material/Module1 Project/2020-movie-analysis


## Import Movie Basics and Ratings From INDB ##

In [79]:
basics_and_ratings = pd.read_csv('Data/imdb.basics_and_ratings_nonulls.csv')
print(basics_and_ratings.shape)
basics_and_ratings.head()

(73856, 8)


title_id                               primary_title  \
0  tt1612608                                 The Italian   
1  tt4874854  The Legend of Zelda: The Triforce Prophecy   
2  tt1718842                          Nakalnya anak muda   
3  tt1645134                                  Sandcastle   
4  tt1645129                                     Planzet   

                               original_title  start_year  runtime_minutes  \
0                                   L'Italien        2010            102.0   
1  The Legend of Zelda: The Triforce Prophecy        2010            120.0   
2                          Nakalnya anak muda        2010             75.0   
3                                  Sandcastle        2010             91.0   
4                                  Puranzetto        2010             53.0   

                      genres  averagerating  numvotes  
0               Comedy,Drama            5.8      1360  
1  Adventure,Fantasy,Romance            4.8         8  
2            Horror,Thriller            4.0        12  
3                      Drama            6.3       116  
4    Action,Animation,Sci-Fi            4.6       290

In [80]:
basics_and_ratings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 8 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   title_id         73856 non-null  object 
 1   primary_title    73856 non-null  object 
 2   original_title   73856 non-null  object 
 3   start_year       73856 non-null  int64  
 4   runtime_minutes  66236 non-null  float64
 5   genres           73052 non-null  object 
 6   averagerating    73856 non-null  float64
 7   numvotes         73856 non-null  int64  
dtypes: float64(2), int64(2), object(4)
memory usage: 4.5+ MB


In [81]:
basics_and_ratings.original_title.nunique()

# there are 73856 unique entries in this table. But there are only  69993 unique primary titles, this means that
# different movies share the same primary title. Only 71097 original titles, as well. 

71097

In [82]:
# information from 2010 - 2019
basics_and_ratings.start_year.value_counts()

2016    8721
2017    8713
2015    8494
2014    8371
2013    7990
2012    7680
2018    7526
2011    7389
2010    6792
2019    2180
Name: start_year, dtype: int64

### clean up null values ###

In [83]:
basics_and_ratings.isnull().sum()

# runtime has a variety of null values, but our questions don't require runtime so best to drop the column

title_id              0
primary_title         0
original_title        0
start_year            0
runtime_minutes    7620
genres              804
averagerating         0
numvotes              0
dtype: int64

In [84]:
b_and_r_cleaned = basics_and_ratings.drop('runtime_minutes', axis = 1)
b_and_r_cleaned

title_id                               primary_title  \
0       tt1612608                                 The Italian   
1       tt4874854  The Legend of Zelda: The Triforce Prophecy   
2       tt1718842                          Nakalnya anak muda   
3       tt1645134                                  Sandcastle   
4       tt1645129                                     Planzet   
...           ...                                         ...   
73851  tt10078894                           Deadly Playthings   
73852   tt6643972                               Triple Threat   
73853   tt7936092     Black Snake: La légende du serpent noir   
73854  tt10080924                                     Sweater   
73855   tt9916160                                  Drømmeland   

                                   original_title  start_year  \
0                                       L'Italien        2010   
1      The Legend of Zelda: The Triforce Prophecy        2010   
2                              Nakalnya anak muda        2010   
3                                      Sandcastle        2010   
4                                      Puranzetto        2010   
...                                           ...         ...   
73851                           Deadly Playthings        2019   
73852                               Triple Threat        2019   
73853     Black Snake: La légende du serpent noir        2019   
73854                                     Sweater        2019   
73855                                  Drømmeland        2019   

                          genres  averagerating  numvotes  
0                   Comedy,Drama            5.8      1360  
1      Adventure,Fantasy,Romance            4.8         8  
2                Horror,Thriller            4.0        12  
3                          Drama            6.3       116  
4        Action,Animation,Sci-Fi            4.6       290  
...                          ...            ...       ...  
73851                     Horror            2.3        30  
73852            Action,Thriller            5.5      4812  
73853                     Comedy            4.9        62  
73854                      Drama            9.2        79  
73855                Documentary            6.5        11  

[73856 rows x 7 columns]

In [88]:
# still need to figure out what to do null values for genres. seems that they are foreign films, do we want to keep
# these? or drop them?
b_and_r_cleaned[b_and_r_cleaned.genres.isnull()]

title_id                                primary_title  \
32      tt1645166                                       Torino   
240     tt1649393                                      Piedras   
333     tt1621833                           My American Exodus   
342     tt3678082  Min pinlige familie og mutantdræbersneglene   
374     tt2007453                                       Sailor   
...           ...                                          ...   
72559   tt9251858                                        Hippi   
72837  tt10161980                             Ituzaingo V3rit4   
72887  tt10161248                            Método Livingston   
73637  tt10122528                                   Miel-Emile   
73755   tt7907676                                    Alexandra   

                                    original_title  year genres  \
32                                          Torino  2010    NaN   
240                                        Piedras  2010    NaN   
333                             My American Exodus  2010    NaN   
342    Min pinlige familie og mutantdræbersneglene  2010    NaN   
374                                         Sailor  2010    NaN   
...                                            ...   ...    ...   
72559                                        Hippi  2019    NaN   
72837                             Ituzaingo V3rit4  2019    NaN   
72887                            Método Livingston  2019    NaN   
73637                                   Miel-Emile  2019    NaN   
73755                                   Aleksandra  2019    NaN   

       averagerating  numvotes  
32               7.1        12  
240              4.9         7  
333              7.6         5  
342              8.5         6  
374              7.6        17  
...              ...       ...  
72559            8.2        28  
72837            4.2         5  
72887            8.2         5  
73637            8.2         5  
73755            6.7         7  

[804 rows x 7 columns]

### change start_year to year ###

In [86]:
b_and_r_cleaned.rename(columns = {'start_year' : 'year'}, inplace = True)
print(b_and_r_cleaned.shape)
b_and_r_cleaned.head()

(73856, 7)


title_id                               primary_title  \
0  tt1612608                                 The Italian   
1  tt4874854  The Legend of Zelda: The Triforce Prophecy   
2  tt1718842                          Nakalnya anak muda   
3  tt1645134                                  Sandcastle   
4  tt1645129                                     Planzet   

                               original_title  year  \
0                                   L'Italien  2010   
1  The Legend of Zelda: The Triforce Prophecy  2010   
2                          Nakalnya anak muda  2010   
3                                  Sandcastle  2010   
4                                  Puranzetto  2010   

                      genres  averagerating  numvotes  
0               Comedy,Drama            5.8      1360  
1  Adventure,Fantasy,Romance            4.8         8  
2            Horror,Thriller            4.0        12  
3                      Drama            6.3       116  
4    Action,Animation,Sci-Fi            4.6       290

## Import Box Office Mojo Data Set ##

In [91]:
bom_gross = pd.read_csv('Data/bom.movie_gross.csv', thousands = ',') # thousands parameter gets rid of commas 
print(bom_gross.shape)
bom_gross.head()

(3387, 5)


title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

   foreign_gross  year  
0    652000000.0  2010  
1    691300000.0  2010  
2    664300000.0  2010  
3    535700000.0  2010  
4    513900000.0  2010

In [92]:
# information from years 2010 - 2018
bom_gross.year.value_counts()

2015    450
2016    436
2012    400
2011    399
2014    395
2013    350
2010    328
2017    321
2018    308
Name: year, dtype: int64

### clean null values ###

In [93]:
# check for null values 
bom_gross.isnull().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [94]:
# fill nulls in domestic and foreign gross with 0's
bom_gross.domestic_gross = bom_gross.domestic_gross.fillna(0)
bom_gross.foreign_gross = bom_gross.foreign_gross.fillna(0)
bom_gross.isnull().sum()

title             0
studio            5
domestic_gross    0
foreign_gross     0
year              0
dtype: int64

In [95]:
# check null values for studio, since there is only 5 with null values just drop those rows
bom_gross[bom_gross.studio.isnull()]

title studio  domestic_gross  foreign_gross  \
210   Outside the Law (Hors-la-loi)    NaN         96900.0      3300000.0   
555         Fireflies in the Garden    NaN         70600.0      3300000.0   
933           Keith Lemon: The Film    NaN             0.0      4000000.0   
1862                 Plot for Peace    NaN          7100.0            0.0   
2825               Secret Superstar    NaN             0.0    122000000.0   

      year  
210   2010  
555   2011  
933   2012  
1862  2014  
2825  2017

In [96]:
bom_gross = bom_gross.dropna(axis = 0, subset = ['studio'])

In [97]:
# successfully dropped null values
bom_gross.isnull().sum()

title             0
studio            0
domestic_gross    0
foreign_gross     0
year              0
dtype: int64

 ### make new column called total_gross that is sum of domestic gross and foreign gross ###


In [98]:
print(bom_gross.domestic_gross.dtype)
print(bom_gross.foreign_gross.dtype)

float64
float64


In [99]:
# make new column called total_gross that is sum of domestic gross and foreign gross
total_gross = bom_gross['domestic_gross'] + bom_gross['foreign_gross']
bom_gross['total_gross'] = total_gross

In [100]:
# would like to figure out a way to change the scientific notation to standard form
bom_gross_cleaned = bom_gross
print(bom_gross_cleaned.shape)
bom_gross_cleaned.head()

(3382, 6)


title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

   foreign_gross  year   total_gross  
0    652000000.0  2010  1.067000e+09  
1    691300000.0  2010  1.025500e+09  
2    664300000.0  2010  9.603000e+08  
3    535700000.0  2010  8.283000e+08  
4    513900000.0  2010  7.526000e+08

In [145]:
bom_gross_cleaned.title.nunique()

3381

## Import The Numbers Data Set ##

In [29]:
budgets = pd.read_csv('Data/tn.movie_budgets.csv')
print(budgets.shape)
budgets.head()

(5782, 6)


id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [30]:
budgets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


### clean up release_date column ###

In [51]:
# extract year and convert it to int64 for one case
example = budgets.release_date[0]
int(example[-4:])

int

In [54]:
# use lambda function to apply to entire column
budgets.release_date = budgets.release_date.apply(lambda x : int(x[-4:]))

In [55]:
budgets.head()

id  release_date                                        movie  \
0   1          2009                                       Avatar   
1   2          2011  Pirates of the Caribbean: On Stranger Tides   
2   3          2019                                 Dark Phoenix   
3   4          2015                      Avengers: Age of Ultron   
4   5          2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [58]:
# rename release date to year, for consistency across tables 
budgets.rename(columns = {'release_date' : 'year'}, inplace = True)

In [60]:
budgets.head()

id  year                                        movie production_budget  \
0   1  2009                                       Avatar      $425,000,000   
1   2  2011  Pirates of the Caribbean: On Stranger Tides      $410,600,000   
2   3  2019                                 Dark Phoenix      $350,000,000   
3   4  2015                      Avengers: Age of Ultron      $330,600,000   
4   5  2017            Star Wars Ep. VIII: The Last Jedi      $317,000,000   

  domestic_gross worldwide_gross  
0   $760,507,625  $2,776,345,279  
1   $241,063,875  $1,045,663,875  
2    $42,762,350    $149,762,350  
3   $459,005,868  $1,403,013,963  
4   $620,181,382  $1,316,721,747

In [62]:
# sort by dataframe by year while we are at it
budgets.sort_values(by = 'year', inplace  = True)
budgets.head()

id  year                           movie production_budget  \
5677  78  1915           The Birth of a Nation          $110,000   
5523  24  1916                     Intolerance          $385,907   
5614  15  1916    20,000 Leagues Under the Sea          $200,000   
5683  84  1920  Over the Hill to the Poorhouse          $100,000   
4569  70  1925   Ben-Hur: A Tale of the Christ        $3,900,000   

     domestic_gross worldwide_gross  
5677    $10,000,000     $11,000,000  
5523             $0              $0  
5614     $8,000,000      $8,000,000  
5683     $3,000,000      $3,000,000  
4569     $9,000,000      $9,000,000

In [119]:
# return dataframe that only has movie data from 2000 - 2020
budgets_cleaned = budgets[budgets.year >= 2000]
print(budgets_cleaned.shape)
budgets_cleaned.head()

(4387, 6)


id  year             movie production_budget domestic_gross  \
5288  89  2000      The Specials        $1,000,000        $12,996   
3308   9  2000         The Claim       $13,000,000       $622,023   
1863  64  2000  Rugrats in Paris       $30,000,000    $76,501,438   
324   25  2000       The Patriot      $110,000,000   $113,330,342   
3891  92  2000   American Psycho        $8,000,000    $15,070,285   

     worldwide_gross  
5288         $12,996  
3308      $1,375,635  
1863    $103,284,813  
324     $215,300,000  
3891     $34,270,285

### convert production_budget, domest_gross, worldwide_gross to int ##

In [120]:
# extract string, get rid of comma and dollar sign, convert to integer
example = budgets_cleaned.production_budget[0]
int(example.replace('$', '').replace(',', ''))

425000000

In [121]:
# define function 
def string_to_int(num_str) :
    return int(num_str.replace('$', '').replace(',', ''))

In [124]:
# apply function to all entries in a column
budgets_cleaned.production_budget = budgets_cleaned.production_budget.apply(string_to_int)

/opt/anaconda3/lib/python3.8/site-packages/pandas/core/generic.py:5303: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value


In [128]:
# apply function to remaining columns
budgets_cleaned.domestic_gross = budgets_cleaned.domestic_gross.apply(string_to_int)
budgets_cleaned.worldwide_gross = budgets_cleaned.worldwide_gross.apply(string_to_int)

/opt/anaconda3/lib/python3.8/site-packages/pandas/core/generic.py:5303: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value


In [129]:
budgets_cleaned.head()

id  year             movie  production_budget  domestic_gross  \
5288  89  2000      The Specials            1000000           12996   
3308   9  2000         The Claim           13000000          622023   
1863  64  2000  Rugrats in Paris           30000000        76501438   
324   25  2000       The Patriot          110000000       113330342   
3891  92  2000   American Psycho            8000000        15070285   

      worldwide_gross  
5288            12996  
3308          1375635  
1863        103284813  
324         215300000  
3891         34270285

In [135]:
# rename worldwide_gross to total gross for consistency
budgets_cleaned.rename(columns = {'worldwide_gross' : 'total_gross'}, inplace = True)
budgets_cleaned.head()

/opt/anaconda3/lib/python3.8/site-packages/pandas/core/frame.py:4125: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().rename(


id  year             movie  production_budget  domestic_gross  \
5288  89  2000      The Specials            1000000           12996   
3308   9  2000         The Claim           13000000          622023   
1863  64  2000  Rugrats in Paris           30000000        76501438   
324   25  2000       The Patriot          110000000       113330342   
3891  92  2000   American Psycho            8000000        15070285   

      total_gross  
5288        12996  
3308      1375635  
1863    103284813  
324     215300000  
3891     34270285

### examine null values in the data frame ###

In [133]:
# no null values in the data frame :)
budgets_cleaned.isnull().any()

id                   False
year                 False
movie                False
production_budget    False
domestic_gross       False
worldwide_gross      False
dtype: bool

### determining movie profits ###

In [136]:
# determine profitability for each movie as ratio of total_gross / production_budget
budgets_cleaned['profitability'] = budgets_cleaned['total_gross'] / budgets_cleaned['production_budget']

<ipython-input-136-e71febc83e22>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  budgets_cleaned['profitability'] = budgets_cleaned['total_gross'] / budgets_cleaned['production_budget']


In [147]:
print(budgets_cleaned.shape)
budgets_cleaned.head()

(4387, 7)


id  year             movie  production_budget  domestic_gross  \
5288  89  2000      The Specials            1000000           12996   
3308   9  2000         The Claim           13000000          622023   
1863  64  2000  Rugrats in Paris           30000000        76501438   
324   25  2000       The Patriot          110000000       113330342   
3891  92  2000   American Psycho            8000000        15070285   

      total_gross  profitability  
5288        12996       0.012996  
3308      1375635       0.105818  
1863    103284813       3.442827  
324     215300000       1.957273  
3891     34270285       4.283786

In [138]:
budgets_cleaned.sort_values(by = 'profitability', ascending = False)

id  year                     movie  production_budget  domestic_gross  \
5492   93  2009       Paranormal Activity             450000       107918810   
5679   80  2015               The Gallows             100000        22764410   
5709   10  2004             Super Size Me              65000        11529368   
5781   82  2005         My Date With Drew               1100          181041   
5656   57  2007                      Once             150000         9445857   
...   ...   ...                       ...                ...             ...   
4850   51  2014         Not Safe For Work            2500000               0   
5599  100  2014                Horse Camp             250000               0   
5565   66  2015  Amidst the Devil's Wings             300000               0   
5640   41  2015            Perfect Cowboy             200000               0   
535    36  2020          Call of the Wild           82000000               0   

      total_gross  profitability  
5492    194183034     431.517853  
5679     41656474     416.564740  
5709     22233808     342.058585  
5781       181041     164.582727  
5656     23323631     155.490873  
...           ...            ...  
4850            0       0.000000  
5599            0       0.000000  
5565            0       0.000000  
5640            0       0.000000  
535             0       0.000000  

[4387 rows x 7 columns]

In [139]:
budgets_cleaned.sort_values(by = 'production_budget', ascending = False)

id  year                                        movie  \
0      1  2009                                       Avatar   
1      2  2011  Pirates of the Caribbean: On Stranger Tides   
2      3  2019                                 Dark Phoenix   
3      4  2015                      Avengers: Age of Ultron   
4      5  2017            Star Wars Ep. VIII: The Last Jedi   
...   ..   ...                                          ...   
5774  75  2004                                       Primer   
5776  77  2004                              The Mongol King   
5779  80  2005                Return to the Land of Wonders   
5780  81  2015                         A Plague So Pleasant   
5781  82  2005                            My Date With Drew   

      production_budget  domestic_gross  total_gross  profitability  
0             425000000       760507625   2776345279       6.532577  
1             410600000       241063875   1045663875       2.546673  
2             350000000        42762350    149762350       0.427892  
3             330600000       459005868   1403013963       4.243841  
4             317000000       620181382   1316721747       4.153696  
...                 ...             ...          ...            ...  
5774               7000          424760       841926     120.275143  
5776               7000             900          900       0.128571  
5779               5000            1338         1338       0.267600  
5780               1400               0            0       0.000000  
5781               1100          181041       181041     164.582727  

[4387 rows x 7 columns]

In [143]:
budgets_cleaned.sort_values(by = 'profitability')

id  year                movie  production_budget  domestic_gross  \
535   36  2020     Call of the Wild           82000000               0   
4565  66  2012  How to Fall in Love            4000000               0   
5330  31  2012               Trance             950000               0   
5311  12  2012         Truth or Die            1000000               0   
5427  28  2012            High Road             546173               0   
...   ..   ...                  ...                ...             ...   
5656  57  2007                 Once             150000         9445857   
5781  82  2005    My Date With Drew               1100          181041   
5709  10  2004        Super Size Me              65000        11529368   
5679  80  2015          The Gallows             100000        22764410   
5492  93  2009  Paranormal Activity             450000       107918810   

      total_gross  profitability  
535             0       0.000000  
4565            0       0.000000  
5330            0       0.000000  
5311            0       0.000000  
5427            0       0.000000  
...           ...            ...  
5656     23323631     155.490873  
5781       181041     164.582727  
5709     22233808     342.058585  
5679     41656474     416.564740  
5492    194183034     431.517853  

[4387 rows x 7 columns]

In [144]:
budgets_cleaned[budgets_cleaned.profitability != 0].sort_values(by = 'profitability')

id  year                movie  production_budget  domestic_gross  \
5302   3  2015        American Hero            1000000               0   
4081  82  2010   Perrier’s Bounty            6600000             828   
3818  19  2015           Skin Trade            9000000            1242   
5335  36  2014          Food Chains             913000               0   
3161  62  2015        Broken Horses           15000000               0   
...   ..   ...                  ...                ...             ...   
5656  57  2007                 Once             150000         9445857   
5781  82  2005    My Date With Drew               1100          181041   
5709  10  2004        Super Size Me              65000        11529368   
5679  80  2015          The Gallows             100000        22764410   
5492  93  2009  Paranormal Activity             450000       107918810   

      total_gross  profitability  
5302           26       0.000026  
4081          828       0.000125  
3818         1242       0.000138  
5335          176       0.000193  
3161         3471       0.000231  
...           ...            ...  
5656     23323631     155.490873  
5781       181041     164.582727  
5709     22233808     342.058585  
5679     41656474     416.564740  
5492    194183034     431.517853  

[4045 rows x 7 columns]

## Export Cleaned Data to New Folder ##

In [148]:
!pwd

/Users/italo/Desktop/Course Material/Module1 Project/2020-movie-analysis


In [149]:
b_and_r_cleaned.to_csv('Cleaned Data/cleaned_imdb-basics-and-ratings.csv')
bom_gross_cleaned.to_csv('Cleaned Data/cleaned_bom-movie-gross.csv')
budgets_cleaned.to_csv('Cleaned Data/cleaned_tn-movie-budgets.csv')

## Cleaned Data Sets Summary ##

**b_and_r_cleaned** gives title, year, genre, ratings, and number of votes for each movie. It has unique 73857 entries though there are many duplicate movie titles in the data frame, i.e. movies that share the same name. This data set can be used to for a ratings analysis for individual movies. Movies are from 2010 - 2019.

**bom_gross_cleaned** gives title, production studio, year, domestic gross, foreign gross, and total gross for each movie. There are 3382 entries in the data frame though only 3381 titles are unique (could be a duplicate?). This data set is good for analyzing yearly profits aggregated by studio. Can also be used to determine a studios yearly market share and trends over time. Can be used to determine profits over time, e.g. Movies are from 2010 - 2018.

**budgets_cleaned** gives year, title, production budget, domestic gross, total gross, and profitability for each movie. It has 4387 entries in the data frame. However, there are only 4364 unique movie titles implying that either movies have the same name or are duplicates. It can be used to see correlation between production budget and total gross or production budget and profitability. Movies are from 2000 - 2020. As such, this data set can be used in our analysis of 2008 recession and the correlation between unemployment rate and profitability of movies. 

In [3]:
import pandas as pd

In [4]:
b_and_r_cleaned = pd.read_csv('Cleaned Data/cleaned_imdb-basics-and-ratings.csv')
print(b_and_r_cleaned.shape)
b_and_r_cleaned.head()

(73856, 8)


Unnamed: 0   title_id                               primary_title  \
0           0  tt1612608                                 The Italian   
1           1  tt4874854  The Legend of Zelda: The Triforce Prophecy   
2           2  tt1718842                          Nakalnya anak muda   
3           3  tt1645134                                  Sandcastle   
4           4  tt1645129                                     Planzet   

                               original_title  year  \
0                                   L'Italien  2010   
1  The Legend of Zelda: The Triforce Prophecy  2010   
2                          Nakalnya anak muda  2010   
3                                  Sandcastle  2010   
4                                  Puranzetto  2010   

                      genres  averagerating  numvotes  
0               Comedy,Drama            5.8      1360  
1  Adventure,Fantasy,Romance            4.8         8  
2            Horror,Thriller            4.0        12  
3                      Drama            6.3       116  
4    Action,Animation,Sci-Fi            4.6       290

In [5]:
len(b_and_r_cleaned.title_id)

73856

In [6]:
bom_gross_cleaned = pd.read_csv('Cleaned Data/cleaned_bom-movie-gross.csv')
print(bom_gross_cleaned.shape)
bom_gross_cleaned.head()

(3382, 7)


Unnamed: 0                                        title studio  \
0           0                                  Toy Story 3     BV   
1           1                   Alice in Wonderland (2010)     BV   
2           2  Harry Potter and the Deathly Hallows Part 1     WB   
3           3                                    Inception     WB   
4           4                          Shrek Forever After   P/DW   

   domestic_gross  foreign_gross  year   total_gross  
0     415000000.0    652000000.0  2010  1.067000e+09  
1     334200000.0    691300000.0  2010  1.025500e+09  
2     296000000.0    664300000.0  2010  9.603000e+08  
3     292600000.0    535700000.0  2010  8.283000e+08  
4     238700000.0    513900000.0  2010  7.526000e+08

In [7]:
budgets_cleaned = pd.read_csv('Cleaned Data/cleaned_tn-movie-budgets.csv')
print(budgets_cleaned.shape)
budgets_cleaned.head()

(4387, 8)


Unnamed: 0  id  year             movie  production_budget  domestic_gross  \
0        5288  89  2000      The Specials            1000000           12996   
1        3308   9  2000         The Claim           13000000          622023   
2        1863  64  2000  Rugrats in Paris           30000000        76501438   
3         324  25  2000       The Patriot          110000000       113330342   
4        3891  92  2000   American Psycho            8000000        15070285   

   total_gross  profitability  
0        12996       0.012996  
1      1375635       0.105818  
2    103284813       3.442827  
3    215300000       1.957273  
4     34270285       4.283786

In [8]:
budgets_cleaned[budgets_cleaned.movie == 'Toy Story 3']

Unnamed: 0  id  year        movie  production_budget  domestic_gross  \
2315          46  47  2010  Toy Story 3          200000000       415004880   

      total_gross  profitability  
2315   1068879522       5.344398

# exploding data #

In [9]:
names = pd.read_csv('Data/imdb.name.basics.csv')
print(names.shape)
names.head()

(606648, 6)


nconst       primary_name  birth_year  death_year  \
0  nm0061671  Mary Ellen Bauder         NaN         NaN   
1  nm0061865       Joseph Bauer         NaN         NaN   
2  nm0062070         Bruce Baum         NaN         NaN   
3  nm0062195       Axel Baumann         NaN         NaN   
4  nm0062798        Pete Baxter         NaN         NaN   

                                 primary_profession  \
0         miscellaneous,production_manager,producer   
1        composer,music_department,sound_department   
2                        miscellaneous,actor,writer   
3  camera_department,cinematographer,art_department   
4  production_designer,art_department,set_decorator   

                          known_for_titles  
0  tt0837562,tt2398241,tt0844471,tt0118553  
1  tt0896534,tt6791238,tt0287072,tt1682940  
2  tt1470654,tt0363631,tt0104030,tt0102898  
3  tt0114371,tt2004304,tt1618448,tt1224387  
4  tt0452644,tt0452692,tt3458030,tt2178256

In [10]:
principals = pd.read_csv('Data/imdb.title.principals.csv')
print(principals.shape)
principals.head()

(1028186, 6)


tconst  ordering     nconst  category       job        characters
0  tt0111414         1  nm0246005     actor       NaN       ["The Man"]
1  tt0111414         2  nm0398271  director       NaN               NaN
2  tt0111414         3  nm3739909  producer  producer               NaN
3  tt0323808        10  nm0059247    editor       NaN               NaN
4  tt0323808         1  nm3579312   actress       NaN  ["Beth Boothby"]

In [11]:
principals.nconst.nunique()

604546

In [12]:
principals.tconst.nunique()

143454

In [13]:
casts = pd.merge(principals, names, on = 'nconst')
casts.head()

tconst  ordering     nconst  category       job   characters  \
0  tt0111414         1  nm0246005     actor       NaN  ["The Man"]   
1  tt0111414         2  nm0398271  director       NaN          NaN   
2  tt5573596         5  nm0398271  director       NaN          NaN   
3  tt0111414         3  nm3739909  producer  producer          NaN   
4  tt0323808        10  nm0059247    editor       NaN          NaN   

            primary_name  birth_year  death_year  \
0           Tommy Dysart         NaN         NaN   
1           Frank Howson      1952.0         NaN   
2           Frank Howson      1952.0         NaN   
3  Barry Porter-Robinson         NaN         NaN   
4            Sean Barton      1944.0         NaN   

                               primary_profession  \
0                                           actor   
1                           actor,writer,producer   
2                           actor,writer,producer   
3                         producer,art_department   
4  editor,editorial_department,assistant_director   

                          known_for_titles  
0  tt0093120,tt0076974,tt0084296,tt0077064  
1  tt0104271,tt0094789,tt0102076,tt0111414  
2  tt0104271,tt0094789,tt0102076,tt0111414  
3  tt0290884,tt0101374,tt0111414,tt1566940  
4  tt0402910,tt1022883,tt0086190,tt0490181

In [14]:
casts.drop(labels = ['ordering', 'characters', 'job', 'birth_year', 'death_year', 
                     'primary_profession', 'known_for_titles'], axis = 1, inplace = True)

In [15]:
casts.head()

tconst     nconst  category           primary_name
0  tt0111414  nm0246005     actor           Tommy Dysart
1  tt0111414  nm0398271  director           Frank Howson
2  tt5573596  nm0398271  director           Frank Howson
3  tt0111414  nm3739909  producer  Barry Porter-Robinson
4  tt0323808  nm0059247    editor            Sean Barton

In [16]:
casts.rename(columns = {'tconst' : 'title_id'}, inplace = True)


## merging both imdb data sets ##

In [17]:
print(casts.shape)
casts.head()

(1027912, 4)


title_id     nconst  category           primary_name
0  tt0111414  nm0246005     actor           Tommy Dysart
1  tt0111414  nm0398271  director           Frank Howson
2  tt5573596  nm0398271  director           Frank Howson
3  tt0111414  nm3739909  producer  Barry Porter-Robinson
4  tt0323808  nm0059247    editor            Sean Barton

In [18]:
print(b_and_r_cleaned.shape)
b_and_r_cleaned.head()

(73856, 8)


Unnamed: 0   title_id                               primary_title  \
0           0  tt1612608                                 The Italian   
1           1  tt4874854  The Legend of Zelda: The Triforce Prophecy   
2           2  tt1718842                          Nakalnya anak muda   
3           3  tt1645134                                  Sandcastle   
4           4  tt1645129                                     Planzet   

                               original_title  year  \
0                                   L'Italien  2010   
1  The Legend of Zelda: The Triforce Prophecy  2010   
2                          Nakalnya anak muda  2010   
3                                  Sandcastle  2010   
4                                  Puranzetto  2010   

                      genres  averagerating  numvotes  
0               Comedy,Drama            5.8      1360  
1  Adventure,Fantasy,Romance            4.8         8  
2            Horror,Thriller            4.0        12  
3                      Drama            6.3       116  
4    Action,Animation,Sci-Fi            4.6       290

In [19]:
merged_imdb = pd.merge(b_and_r_cleaned, casts, on = 'title_id')

In [20]:
print(merged_imdb.shape)
merged_imdb.tail(20)

(629598, 11)


Unnamed: 0    title_id                            primary_title  \
629578       73853   tt7936092  Black Snake: La légende du serpent noir   
629579       73853   tt7936092  Black Snake: La légende du serpent noir   
629580       73853   tt7936092  Black Snake: La légende du serpent noir   
629581       73854  tt10080924                                  Sweater   
629582       73854  tt10080924                                  Sweater   
629583       73854  tt10080924                                  Sweater   
629584       73854  tt10080924                                  Sweater   
629585       73854  tt10080924                                  Sweater   
629586       73854  tt10080924                                  Sweater   
629587       73854  tt10080924                                  Sweater   
629588       73854  tt10080924                                  Sweater   
629589       73854  tt10080924                                  Sweater   
629590       73854  tt10080924                                  Sweater   
629591       73855   tt9916160                               Drømmeland   
629592       73855   tt9916160                               Drømmeland   
629593       73855   tt9916160                               Drømmeland   
629594       73855   tt9916160                               Drømmeland   
629595       73855   tt9916160                               Drømmeland   
629596       73855   tt9916160                               Drømmeland   
629597       73855   tt9916160                               Drømmeland   

                                 original_title  year       genres  \
629578  Black Snake: La légende du serpent noir  2019       Comedy   
629579  Black Snake: La légende du serpent noir  2019       Comedy   
629580  Black Snake: La légende du serpent noir  2019       Comedy   
629581                                  Sweater  2019        Drama   
629582                                  Sweater  2019        Drama   
629583                                  Sweater  2019        Drama   
629584                                  Sweater  2019        Drama   
629585                                  Sweater  2019        Drama   
629586                                  Sweater  2019        Drama   
629587                                  Sweater  2019        Drama   
629588                                  Sweater  2019        Drama   
629589                                  Sweater  2019        Drama   
629590                                  Sweater  2019        Drama   
629591                               Drømmeland  2019  Documentary   
629592                               Drømmeland  2019  Documentary   
629593                               Drømmeland  2019  Documentary   
629594                               Drømmeland  2019  Documentary   
629595                               Drømmeland  2019  Documentary   
629596                               Drømmeland  2019  Documentary   
629597                               Drømmeland  2019  Documentary   

        averagerating  numvotes      nconst         category  \
629578            4.9        62   nm9365887         composer   
629579            4.9        62   nm1169001            actor   
629580            4.9        62  nm10472428           writer   
629581            9.2        79   nm2756056          actress   
629582            9.2        79   nm5088926         producer   
629583            9.2        79  nm10587245         composer   
629584            9.2        79   nm1315113          actress   
629585            9.2        79   nm6684661          actress   
629586            9.2        79   nm1425584            actor   
629587            9.2        79   nm7127455         director   
629588            9.2        79   nm7717308         producer   
629589            9.2        79  nm10594964         producer   
629590            9.2        79   nm7717309         producer   
629591            6.5        11   nm0462955           editor   
6

In [21]:
merged_imdb.title_id.nunique()

73685

In [22]:
b_and_r_cleaned.title_id.nunique()

73856

In [23]:
# return all movies with that 
merged_imdb[merged_imdb.primary_name == 'Oscar Isaac'].sort_values(by = 'averagerating')

Unnamed: 0   title_id                                   primary_title  \
180534       21293  tt2053423                              Revenge for Jolly!   
386993       45675  tt2322517                                          Mojave   
541427       63781  tt0491175                                      Suburbicon   
452941       53453  tt4776998                                     The Promise   
318737       37671  tt1976000                        The Two Faces of January   
246608       29151  tt0409379                                       In Secret   
106680       12686  tt1536048                                            W.E.   
162285       19199  tt1870529                                 Won't Back Down   
578543       68204  tt5208252                                Operation Finale   
618613       72625  tt1488606                                 Triple Frontier   
587725       69208  tt5989218                                     Life Itself   
184519       21756  tt1566501  For Greater Glory: The True Story of Cristiada   
585376       68947  tt6938828                              At Eternity's Gate   
301425       35767  tt2937898                             A Most Violent Year   
347764       41112  tt5001130                                  Time to Choose   
228603       27168  tt2042568                             Inside Llewyn Davis   
278193       32930  tt0470752                                      Ex Machina   
353092       41757  tt2488496      Star Wars: Episode VII - The Force Awakens   

                                        original_title  year  \
180534                              Revenge for Jolly!  2012   
386993                                          Mojave  2015   
541427                                      Suburbicon  2017   
452941                                     The Promise  2016   
318737                        The Two Faces of January  2014   
246608                                       In Secret  2013   
106680                                            W.E.  2011   
162285                                 Won't Back Down  2012   
578543                                Operation Finale  2018   
618613                                 Triple Frontier  2019   
587725                                     Life Itself  2018   
184519  For Greater Glory: The True Story of Cristiada  2012   
585376                              At Eternity's Gate  2018   
301425                             A Most Violent Year  2014   
347764                                  Time to Choose  2015   
228603                             Inside Llewyn Davis  2013   
278193                                      Ex Machina  2014   
353092      Star Wars: Episode VII - The Force Awakens  2015   

                          genres  averagerating  numvotes     nconst category  \
180534              Comedy,Drama            4.6      1202  nm1209966    actor   
386993            Drama,Thriller            5.2      3726  nm1209966    actor   
541427        Comedy,Crime,Drama            5.6     31294  nm1209966    actor   
452941             Drama,History            6.1    164555  nm1209966    actor   
318737          Romance,Thriller            6.2     27542  nm1209966    actor   
246608      Crime,Drama,Thriller            6.2      7045  nm1209966    actor   
106680     Drama,History,Romance            6.3     11314  nm1209966    actor   
162285                     Drama            6.4      5915  nm1209966    actor   
578543   Biography,Drama,History            6.5     20927  nm1209966    actor   
618613    Action,Adventure,Crime            6.5     72202  nm1209966    actor   
587725             Drama,Romance            6.6      9301  nm1209966    actor   
184519         Drama,History,War            6.6      4202  nm1209966    actor   
585376           Biography,Drama            6.9     16558  nm1209966    actor   
301425      Crime,Drama,Thriller            7.0     60165  nm1209966    actor   
347764               Documentary    

In [24]:
merged_imdb[merged_imdb.title_id ==  'tt2488496']

Unnamed: 0   title_id                               primary_title  \
353091       41757  tt2488496  Star Wars: Episode VII - The Force Awakens   
353092       41757  tt2488496  Star Wars: Episode VII - The Force Awakens   
353093       41757  tt2488496  Star Wars: Episode VII - The Force Awakens   
353094       41757  tt2488496  Star Wars: Episode VII - The Force Awakens   
353095       41757  tt2488496  Star Wars: Episode VII - The Force Awakens   
353096       41757  tt2488496  Star Wars: Episode VII - The Force Awakens   
353097       41757  tt2488496  Star Wars: Episode VII - The Force Awakens   
353098       41757  tt2488496  Star Wars: Episode VII - The Force Awakens   
353099       41757  tt2488496  Star Wars: Episode VII - The Force Awakens   
353100       41757  tt2488496  Star Wars: Episode VII - The Force Awakens   

                                    original_title  year  \
353091  Star Wars: Episode VII - The Force Awakens  2015   
353092  Star Wars: Episode VII - The Force Awakens  2015   
353093  Star Wars: Episode VII - The Force Awakens  2015   
353094  Star Wars: Episode VII - The Force Awakens  2015   
353095  Star Wars: Episode VII - The Force Awakens  2015   
353096  Star Wars: Episode VII - The Force Awakens  2015   
353097  Star Wars: Episode VII - The Force Awakens  2015   
353098  Star Wars: Episode VII - The Force Awakens  2015   
353099  Star Wars: Episode VII - The Force Awakens  2015   
353100  Star Wars: Episode VII - The Force Awakens  2015   

                          genres  averagerating  numvotes     nconst  \
353091  Action,Adventure,Fantasy            8.0    784780  nm5397459   
353092  Action,Adventure,Fantasy            8.0    784780  nm1209966   
353093  Action,Adventure,Fantasy            8.0    784780  nm1578335   
353094  Action,Adventure,Fantasy            8.0    784780  nm0009190   
353095  Action,Adventure,Fantasy            8.0    784780  nm1333357   
353096  Action,Adventure,Fantasy            8.0    784780  nm1727304   
353097  Action,Adventure,Fantasy            8.0    784780  nm0005086   
353098  Action,Adventure,Fantasy            8.0    784780  nm0000184   
353099  Action,Adventure,Fantasy            8.0    784780  nm0001410   
353100  Action,Adventure,Fantasy            8.0    784780  nm3915784   

        category      primary_name  
353091   actress      Daisy Ridley  
353092     actor       Oscar Isaac  
353093    writer     Michael Arndt  
353094  director       J.J. Abrams  
353095  producer        Bryan Burk  
353096     actor  Domhnall Gleeson  
353097  producer  Kathleen Kennedy  
353098    writer      George Lucas  
353099    writer   Lawrence Kasdan  
353100     actor       John Boyega

## export new merged imdb data set ##

In [25]:
casts.to_csv('Cleaned Data/imdb.movies_basics_and_crew.csv')


## clean budgets for unemployment analysis ##

In [26]:
budgets = pd.read_csv('Data/tn.movie_budgets.csv')
print(budgets.shape)
budgets.head()

(5782, 6)


id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [27]:
# clean money value data and make them numeric
def string_to_int(num_str) :
    return int(num_str.replace('$', '').replace(',', ''))

In [29]:
budgets.production_budget = budgets.production_budget.apply(string_to_int)
budgets.domestic_gross = budgets.domestic_gross.apply(string_to_int)
budgets.worldwide_gross = budgets.worldwide_gross.apply(string_to_int)

In [31]:
budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

   production_budget  domestic_gross  worldwide_gross  
0          425000000       760507625       2776345279  
1          410600000       241063875       1045663875  
2          350000000        42762350        149762350  
3          330600000       459005868       1403013963  
4          317000000       620181382       1316721747

In [32]:
budgets_unempl = budgets

In [33]:
budgets_unempl.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

   production_budget  domestic_gross  worldwide_gross  
0          425000000       760507625       2776345279  
1          410600000       241063875       1045663875  
2          350000000        42762350        149762350  
3          330600000       459005868       1403013963  
4          317000000       620181382       1316721747

In [35]:
example = budgets_unempl['release_date'][0]

In [42]:
example

'Dec 18, 2009'